src/api.ts

```diff
@@ -0,0 +1,55 @@
+import axios, { AxiosResponse } from "axios";
+
+axios.defaults.withCredentials = true;
+
+export const axiosWithCredentials = axios;
+export default axios;
+
+// 定義 api baseUrl
+// TODO: 使用 env 定義不同環境 baseUrl
+const backendRequest = axios.create({
+  baseURL: "http://localhost:8000/api/v1/",
+});
+
+// signIn Api
+interface SignInFormParams {
+  user: {
+    email: string;
+    password: string;
+  };
+}
+
+interface SignInResponse {
+  status: string;
+  message: string;
+}
+
+type ApiUserSignIn = (
+  data: SignInFormParams
+) => Promise<AxiosResponse<SignInResponse>>;
+
+export const apiUserSignIn: ApiUserSignIn = async (data) =>
+  await backendRequest.post<SignInResponse>("/users/sign_in", data);
```

Comment:
沒記醋的話 axios.post 的回傳值的 type 原本就是 Promise 了，不用再用 async await 的語法

```suggestion
// ApiUserSignIn 會是 ()=>Promise<...>
export const apiUserSignIn: ApiUserSignIn = () =>
  backendRequest.post<SignInResponse>("/users/sign_in", data);
```

Reply:
原來如此....
Q: 所以有加跟沒加在這裡是一模一樣？

Reply:
但 linter 好像就是因為知道回傳的是 promise 所以會幫我加 @@?
![image](https://user-images.githubusercontent.com/50241297/193604399-a81cac93-0619-4ef2-99af-f4e021783c47.png)

Reply:
喔喔你需要 async 保留字 但不用 await

Reply:
可是 linter 跟我說一定要 await? 我看 h4b 好像也沒有特別取消這個 linter 🤔 

<img width="905" alt="image" src="https://user-images.githubusercontent.com/50241297/193615943-6f2f31a0-000c-44b6-92d8-aa3e906c15ae.png">

Reply:
這個比較像是避免問題的 lint rule，因為不回傳 await 等於後面的邏輯要預設 data 可能還沒回來
通常一個 api call 後續都會對資料進行處理，所以沒有 await 會出事

Reply:
但如果後面沒有要做其他事情其實不一定要回傳
另外你的 backendRequest.post 回傳的型別難道不是 Promise 嗎？